src/btree/bt_sync_obsolete.c

```diff
@@ -477,8 +479,16 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
         if (ret == 0)
             newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)value.val);
         WT_RET_NOTFOUND_OK(ret);
+        ret = __wt_config_subgets(session, &cval, "addr", &value);
+        if (ret == 0)
+            addr_size = value.len;
```

Comment:
Do we need to keep looping? Should we `break`?

Reply:
No, we need to check all the checkpoints that existed for this table before making a decision.

Reply:
I see, thank you. Can we change the `for` loop to:
```
        while ((ret = __wt_config_next(&ckptconf, &key, &cval)) == 0) {
...
        }
WT_RET_NOTFOUND_OK(ret);
```
See an example [here](https://github.com/wiredtiger/wiredtiger/blob/5a8e337c9a79390d6be8643c7c1c2cb726ad41da/src/config/config_api.c#L222-L265).
I am suggesting this as we are currently not catching any potential error from `__wt_config_next`.

Reply:
Fixed.